pyEpiabm/pyEpiabm/tests/test_sweep/test_host_progression_sweep.py

```diff
@@ -176,6 +189,138 @@ def test_icdf_exception_raise(self):
         test_sweep.transition_time_matrix.loc[row_index, column_index].\
             icdf_choose_noexp.assert_called_once()
 
+    def test_infectiousness_progression(self):
+        """Tests that the output is a numpy ndarray and that the tail of the
+        array is 0, starting at the first element after the last infectious
+        time step, when the number of time steps per day is 1 and the
+        infectious period is 14 days.
+        """
+        with mock.patch('pyEpiabm.Parameters.instance') as mock_param:
+            mock_param.return_value.time_steps_per_day = 1
+            mock_param.return_value.asympt_infect_period = 14
+            # Parameters to determine where the tail starts, has to be the same
+            # as for the parameters called in HostProgressionSweep.
+            infectious_period = pe.Parameters.instance().asympt_infect_period
+            model_time_step = 1 / pe.Parameters.instance().time_steps_per_day
+            num_infectious_ts = int(np.ceil(infectious_period /
+                                            model_time_step))
+            # Initialisation
+            test_sweep = pe.sweep.HostProgressionSweep()
+            infect_prog = test_sweep.infectiousness_progression
+            # Checks output type is numpy array
+            self.assertIsInstance(infect_prog, np.ndarray)
+            # Checks elements are 0 after k and greater than 0 before
+            tail = infect_prog[num_infectious_ts:2550]
+            zeros = np.zeros(2550-num_infectious_ts)
+            self.assertTrue((tail == zeros).all())
+            self.assertTrue((infect_prog[0:num_infectious_ts] >
+                            np.zeros(num_infectious_ts)).all())
+
+    def test_infectiousness_progression_small_time_steps(self):
+        """Tests that the method workd when there are more than 1 time step
+        per day. Also tests that an assertion error is raised if the model
+        time steps length is too small (ie the number of time steps per day is
+        too big).
+        """
+        with mock.patch('pyEpiabm.Parameters.instance') as mock_param:
+            # Same test as test_infectiousness_progression, but with 100 time
+            # steps per day:
+            mock_param.return_value.time_steps_per_day = 100
+            mock_param.return_value.asympt_infect_period = 14
+            infectious_period = pe.Parameters.instance().asympt_infect_period
+            model_time_step = 1 / pe.Parameters.instance().time_steps_per_day
+            num_infectious_ts = int(np.ceil(infectious_period /
+                                            model_time_step))
+            test_sweep = pe.sweep.HostProgressionSweep()
+            infect_prog = test_sweep.infectiousness_progression
+            # Checks output type is numpy array
+            self.assertIsInstance(infect_prog, np.ndarray)
+            # Checks elements are 0 after k and greater than 0 before
```

Comment:
This new test is really great, but maybe update this comment now we have renamed `k`?